### ISSUE TITLE
[Modal] Modals closed by default appear visible and then fade out on load

### ISSUE THREAD

**Issue by frodesigns** [carbon-design-system/carbon-web-components#694]
## Steps to reproduce the issue

1. Visit https://codesandbox.io/s/zrumj
2. Observe the modal appears briefly on load/after refresh, and then fades out

Clip:
https://user-images.githubusercontent.com/6588808/129775211-0a56ff68-bf0e-4b12-b4c8-a9e9c0e773d4.mp4

**Comment by RobertaJHahn** [comment]
Ignacio and Ari to review.

**Comment by frodesigns** [comment]
> @frodesigns Hey there, looking at your code sandbox example, it seems that there is no code that opens the modal on load.
> 
> Here's a forked version of your example using the `open` attribute on `<bx-modal>` that opens it on load:
> https://codesandbox.io/s/frosty-ramanujan-hts1q?file=/index.html

It is not my sandbox but you are correct, the modal is meant to be closed initially. However, the issue is that it is appearing briefly and fading out on load.

**Comment by IgnacioBecerra** [comment]
@frodesigns I can't seem to replicate that behavior on the sandbox, here's a video of what I see:

https://user-images.githubusercontent.com/24970122/131863004-e1d0836e-fa8c-4534-bb9d-ca992a9a3507.mov

**Comment by IgnacioBecerra** [comment]
That being said, you can try adding this to your styles to ensure this flickering doesn't happen
```
    /* Suppress custom element until styles are loaded */
    bx-modal:not(:defined) {
      display: none;
    }
```

**Comment by frodesigns** [comment]
It happens to me intermittently in both Chrome and Firefox.

Clip:
https://user-images.githubusercontent.com/6588808/129775211-0a56ff68-bf0e-4b12-b4c8-a9e9c0e773d4.mp4

**Comment by RobertaJHahn** [comment]
Closing.  No response from issue author.

**Comment by frodesigns** [comment]
Is there any reason you did not roll the suggested fix into the code for this repository?

**Comment by RobertaJHahn** [comment]
@frodesigns You did not respond to @IgnacioBecerra question 22 days ago.  So the issue was closed.

**Comment by frodesigns** [comment]
Yes apologies for not keeping up, but the issue still stands and seems fixed by his suggestion, which should be rolled into this codebase.

**Comment by rccc** [comment]
I have myself this stange behavior, modal appears brielfy  on page load and i think like @frodesigns that the fix should be in codebase.
